WordPress/Classes/Services/MediaThumbnailService.swift

```diff
@@ -19,24 +19,41 @@ class MediaThumbnailService: LocalCoreDataService {
         return MediaThumbnailService.defaultExportQueue
     }()
 
+    private let coreDataStack: CoreDataStackSwift
+
+    /// The initialiser for Objective-C code.
+    ///
+    /// Using `ContextManager` as the argument becuase `CoreDataStackSwift` is not accessible from Objective-C code.
+    @objc
+    convenience init(contextManager: ContextManager) {
+        self.init(coreDataStack: contextManager)
+    }
+
+    init(coreDataStack: CoreDataStackSwift) {
+        self.coreDataStack = coreDataStack
+    }
+
     /// Generate a URL to a thumbnail of the Media, if available.
     ///
     /// - Parameters:
     ///   - media: The Media object the URL should be a thumbnail of.
     ///   - preferredSize: An ideal size of the thumbnail in points. If `zero`, the maximum dimension of the UIScreen is used.
-    ///   - onCompletion: Completion handler passing the URL once available, or nil if unavailable.
-    ///   - onError: Error handler.
+    ///   - onCompletion: Completion handler passing the URL once available, or nil if unavailable. This closure is called on the `exportQueue`.
+    ///   - onError: Error handler. This closure is called on the `exportQueue`.
     ///
     /// - Note: Images may be downloaded and resized if required, avoid requesting multiple explicit preferredSizes
     ///   as several images could be downloaded, resized, and cached, if there are several variations in size.
     ///
     @objc func thumbnailURL(forMedia media: Media, preferredSize: CGSize, onCompletion: @escaping OnThumbnailURL, onError: OnError?) {
-        managedObjectContext.perform {
+        let context = coreDataStack.mainContext
```

Comment:
Using the main context here because this function only read the `Media` instance, without changing it.

Reply:
Also, there is no time consuming work done in the block that's passed to the `perform` function, all heavy lifting are done in background queues.

Reply:
```suggestion
        // We can use the main context here because we only read the `Media` instance, without changing it, and all
        // the time consuming work is done in background queues.
        let context = coreDataStack.mainContext
```